treeherder/etl/seta.py

```diff
@@ -0,0 +1,210 @@
+import logging
+
+from treeherder.seta.runnable_jobs import RunnableJobsClient
+
+LOG = logging.getLogger(__name__)
+
+
+class Treecodes:
+    """This class contain all the mapping we need in SETA and makes it works"""
+
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        self.jobtypes = []
+        self.jobnames = []
+
+        for job in RunnableJobsClient().query_runnable_jobs(repo_name)['results']:
+            # e.g. web-platform-tests-4
+            # e.g. Ubuntu VM 12.04 x64 mozilla-inbound opt test web-platform-tests-4 OR
+            #      desktop-test-linux64/opt-web-platform-tests-4
+            testtype, job_identifier = self._testtype_job_identifier(job)
+
+            # XXX: This has the danger of falling out of date
```

Comment:
can we file a bug to remind us this can get out of date and explore options to make this better.